### ISSUE TITLE
From address failing local\@foo.com <local@bar.com>

### ISSUE THREAD

**Issue by robsco** [rjbs/Email-Stuffer#46]
I believe the above is a valid address, which I've been using on a project for a while.

Upon upgrading Email::Stuffer from 0.014, this now fails:

Argument contains empty host portion of address at /home/rob/perl5/perlbrew/perls/perl-5.22.0/lib/site_perl/5.22.0/Email/MIME/Encode.pm line 70.

Assuming this is due to the new way you're handling the address fields, at 0.016?

**Comment by pali** [comment]
> I believe the above is a valid address

No, header `From: local\@foo.com <local@bar.com>` is invalid. If you want to include character `@` into phrase then you need to quote it, e.g.: `From: "local@foo.com" <local@bar.com>`. Characters `\` or `@` are not allowed in `atom` as `atext`. For more details look into RFC5322, section 3.4: https://tools.ietf.org/html/rfc5322#section-3.4

If you are unsure for how to correctly put `@` or `\` or any other character into phrase (display-name), then use [Email::Address::XS](https://metacpan.org/pod/Email::Address::XS) module. It takes phrase and email address and properly format it (quote and escape what is needed).

```
$ perl -MEmail::Address::XS -E 'say Email::Address::XS->new(phrase => q(local@foo.com), address => q(local@bar.com))->format()'
"local@foo.com" <local@bar.com>
```
